src/AdvancedHMC.jl

```diff
@@ -34,13 +34,54 @@ include("integrator.jl")
 export Leapfrog, JitteredLeapfrog, TemperedLeapfrog
 
 include("trajectory.jl")
-@deprecate find_good_eps find_good_stepsize
-export EndPointTS, SliceTS, MultinomialTS, 
-       StaticTrajectory, HMCDA, NUTS, 
-       ClassicNoUTurn, GeneralisedNoUTurn, 
-       StrictGeneralisedNoUTurn,
+export Trajectory,
+       FixedNSteps, FixedIntegrationTime,
+       ClassicNoUTurn, GeneralisedNoUTurn, StrictGeneralisedNoUTurn,
+       EndPointTS, SliceTS, MultinomialTS, 
        find_good_stepsize
 
+# Useful defaults
+
+struct NUTS{TS, TC} end
+
+"""
+$(SIGNATURES)
+
+Convenient constructor for the no-U-turn sampler (NUTS).
+This falls back to `Trajectory{TS}(int, TC(args...; kwargs...))` where
+
+- `TS<:Union{MultinomialTS, SliceTS}` is the type for trajectory sampler
+- `TC<:Union{ClassicNoUTurn, GeneralisedNoUTurn, StrictGeneralisedNoUTurn}` is the type for termination criterion.
+
+See [`ClassicNoUTurn`](@ref), [`GeneralisedNoUTurn`](@ref) and [`StrictGeneralisedNoUTurn`](@ref) for details in parameters.
+"""
+NUTS{TS, TC}(int::AbstractIntegrator, args...; kwargs...) where {TS, TC} = 
+    Trajectory{TS}(int, TC(args...; kwargs...))
+NUTS(int::AbstractIntegrator, args...; kwargs...) = 
+    Trajectory{MultinomialTS}(int, GeneralisedNoUTurn(args...; kwargs...))
+NUTS(ϵ::AbstractScalarOrVec{<:Real}) =
+    Trajectory{MultinomialTS}(Leapfrog(ϵ), GeneralisedNoUTurn())
+
+export NUTS
+
+# Deprecations for trajectory.jl
+
+struct AbstractTrajectory <: AbstractProposal end
+
+struct StaticTrajectory{TS} end
+@deprecate StaticTrajectory{TS}(int::AbstractIntegrator, L) where {TS} Trajectory{TS}(int, FixedNSteps(L))
+@deprecate StaticTrajectory(int::AbstractIntegrator, L) Trajectory{EndPointTS}(int, FixedNSteps(L))
+@deprecate StaticTrajectory(ϵ::AbstractScalarOrVec{<:Real}, L) Trajectory{EndPointTS}(Leapfrog(ϵ), FixedNSteps(L))
+
+struct HMCDA{TS} end
+@deprecate HMCDA{TS}(int::AbstractIntegrator, λ) where {TS} Trajectory{TS}(int, FixedIntegrationTime(λ))
+@deprecate HMCDA(int::AbstractIntegrator, λ) Trajectory{MetropolisTS}(int, FixedIntegrationTime(λ))
+@deprecate HMCDA(ϵ::AbstractScalarOrVec{<:Real}, λ) Trajectory{MetropolisTS}(Leapfrog(ϵ), FixedIntegrationTime(λ))
+
+@deprecate find_good_eps find_good_stepsize
+
+export AbstractTrajectory, StaticTrajectory, HMCDA, find_good_eps
```

Comment:
Do we need to export `AbstractTrajectory`?

Reply:
We don't. Let me remove this.